### ISSUE TITLE
Error handling is inadequate

### ISSUE THREAD

**Issue by kevinfoley** [googlesamples/google-signin-unity#99]
We're experiencing issues with Google sign-in in our Unity/Firebase game. Login always fails, with the error:

    SignInException: Exception of type 'Google.GoogleSignIn+SignInException' was thrown.

This is not a useful error message. It only tells us that _something_ went wrong. It tells us nothing whatsoever about what the problem is or how to fix it.

I've quadruple-checked the OAuth web app ID, Firebase settings, etc - everything that gets mentioned when searching this generic error message, but still haven't figured out what's wrong. More detailed error messages would save a lot of time.

**Comment by amine307** [comment]
Set up a project and apply in the API Console, 
this link will benefit you:
https://developers.google.com/drive/android/auth

**Comment by kevinfoley** [comment]
@amine307  Yes, of course we've already done that.

**Comment by Mukikaizoku** [comment]
@kevinfoley I agree, more verbose error reporting would greatly reduce troubleshooting. It was a huge pain to get just the sample project to work.

To get more error information, I used adb logcat to monitor logs off of the sample app loaded into Bluestacks. It still took a lot of troubleshooting, but it gave me enough information to figure out why I was getting the SignInException with Status=Error (My Play Services Resolver wasn't working correctly and so I was getting errors related to .jar libraries not being imported). After that I got Status=DeveloperError and for me the problem was using the `debug.keystore`, which I guess doesn't work? Switching to my app's keystore fixed the DeveloperError for me.

**Comment by talhakaya** [comment]
Did anyone find anything about this? I don't have the "DeveloperError", I just have an "Error", which is even less verbose.

**Comment by charles-mersereau** [comment]
Is there a way of getting more attention for this? I'm experiencing this issue and it's very frustrating to have such an inadequate error log.

Also if anyone has a solution to the problem, that'd be great as well. Have looked through some of the other open issues and nothing has helped so far.
